### ISSUE TITLE
Proposed breadcrumb guidelines

### ISSUE THREAD

**Issue by matthewpaulthomas** [ubuntudesign/vanilla-design#33]
Sometimes it’s appropriate to represent the current page inside its own breadcrumbs. Other times, doing that would needlessly repeat a heading just below.

At the design patterns meeting on Thursday, I volunteered to write proposed guidelines for the [Vanilla docs Breadcrumbs page](https://docs.vanillaframework.io/en/patterns/breadcrumbs) on when+how to use one model vs. another.

Here they are:

> # Breadcrumbs
> You can use the breadcrumbs pattern to show the hierarchy from the front page of a site to the current page. Even if there are multiple ways to navigate to the current page, a canonical hierarchy can help users understand the context, or help them navigate to more general information.
>
> You don’t have to add breadcrumb separators manually, because Vanilla adds them using CSS.
>
> Sometimes breadcrumbs are immediately followed by a [heading](https://docs.vanillaframework.io/en/base/typography) representing the current page:
>> Front page > Second level > Third level >
>> ## Fourth level
>>
```html
<ul class="p-breadcrumbs">
  <li class="p-breadcrumbs__item"><a class="p-breadcrumbs__link" href="#">Front page</a></li>
  <li class="p-breadcrumbs__item"><a class="p-breadcrumbs__link" href="#">Second level</a></li>
  <li class="p-breadcrumbs__item"><a class="p-breadcrumbs__link" href="#">Third level</a></li>
</ul>
<h1>Fourth level</h1>
```
> If your page layout doesn’t allow this, you may choose to represent the current page at the end of the breadcrumbs, as well as (or instead of) a main heading.
>> Front page > Second level > Third level > **Fourth level**
```html
<ul class="p-breadcrumbs">
  <li class="p-breadcrumbs__item"><a class="p-breadcrumbs__link" href="#">Front page</a></li>
  <li class="p-breadcrumbs__item"><a class="p-breadcrumbs__link" href="#">Second level</a></li>
  <li class="p-breadcrumbs__item"><a class="p-breadcrumbs__link" href="#">Third level</a></li>
  <li class="p-breadcrumbs__item">Fourth level</li>
</ul>
```
> If you do choose to represent the current page as the final breadcrumb, then:
> - Do not make it a link. That would mean the page was linking to itself, and it would be followed by a misleading separator.
> - Make sure the text is short enough to fit in the space you’ve allocated to breadcrumbs. This might involve using text that is different from the main heading.

If it’s appropriate to link to external guidelines for more detail, the [Nielsen Norman breadcrumb guidelines](https://www.nngroup.com/articles/breadcrumb-navigation-useful/) are old but good.

The sample code assumes that the Vanilla CSS can select whether or not to add a separator after an item based solely on whether the item contains a link, rather than the current approach of whether the item is not the last one in the list. This would be simplest for authors. It would also help them avoid errors like self-links and missing links (such as our own [vanilla-docs-theme#37](https://github.com/vanilla-framework/vanilla-docs-theme/issues/37)), because those errors would look more obviously wrong during testing. However, I understand that it can’t be achieved with current CSS alone. If Vanilla can’t achieve it using some other mechanism, such as `document.querySelector('li.p-breadcrumbs__item:has(> :link)')`, then there would need to be a separate class for the `<li>` that represents the current page.

**Comment by anthonydillon** [comment]
Collect more examples and keep these guidelines from the breadcrumb documentation.

**Comment by matthewpaulthomas** [comment]
I collected 20 examples of popular sites that use breadcrumbs. These were taken from Wikipedia’s list of most popular sites in the world. Unfortunately I had to skip those in languages where I couldn’t tell how they were using breadcrumbs, and substituted sites for popular UK retailers.

[Breadcrumbs examples.pdf](https://github.com/ubuntudesign/vanilla-design/files/1196755/Breadcrumbs.examples.pdf)

Of these 20, exactly half represent the current page inside its own breadcrumbs.
